diff --git a/actors/enemy/enemy.gd b/actors/enemy/enemy.gd
index dc83e4e..d5f540a 100644
--- a/actors/enemy/enemy.gd
+++ b/actors/enemy/enemy.gd
@@ -73,6 +73,10 @@ func _ready():
 	
 func movement(delta: float):
 
+	if knockback.is_active:
+		velocity += knockback.current_force * delta
+		return
+
 	var target_speed := move_direction * move_speed
 
 	if move_direction != 0:
diff --git a/actors/player/player.gd b/actors/player/player.gd
index 26fb0a1..7c36f62 100644
--- a/actors/player/player.gd
+++ b/actors/player/player.gd
@@ -11,6 +11,7 @@ extends CharacterBody2D
 @onready var vfx_system = $funcions/vfx
 @onready var animation_system: AnimationSystem = $funcions/animations
 @onready var state_machine: StateMachine = $funcions/state_machine
+@onready var knockback: Knockback = $Components/Knockback
 
 func _ready():
 	jump_system.jumped.connect(vfx_system.play_jump)
@@ -23,6 +24,13 @@ func _physics_process(delta):
 		velocity.x,
 		Input.get_axis("move_left","move_rigth") != 0
 	)
+
+	movement.update_stats(flow_system.flow, is_on_floor())
+	jump_system.update_stats(flow_system.flow)
+	dash_system.update_stats(flow_system.flow)
+	slide_system.update_stats(flow_system.flow)
+	wall_jump_system.update_stats(flow_system.flow)
+
 	velocity = dash_system.update(
 		velocity,
 		delta,
@@ -54,7 +62,12 @@ func _physics_process(delta):
 		dash_system.is_dashing
 		or slide_system.is_slide
 		or wall_jump_system.is_walljumping
+		or knockback.is_active
 	)
+
+	if knockback.is_active:
+		velocity += knockback.current_force * delta
+
 	
 	if dash_system.is_dashing:
 		state_machine.set_state(StateMachine.State.DASH)
diff --git a/components/hitbox/hitbox.gd b/components/hitbox/hitbox.gd
index 56dc731..c974f75 100644
--- a/components/hitbox/hitbox.gd
+++ b/components/hitbox/hitbox.gd
@@ -2,6 +2,7 @@ extends Area2D
 class_name Hitbox
 
 @export var damage: int = 10
+@export var knockback_force: float = 300.0
 @onready var shape = $CollisionShape2D
 
 func hitbox_on():
diff --git a/components/hurtbox/hurtbox.gd b/components/hurtbox/hurtbox.gd
index 8150b34..ee8be92 100644
--- a/components/hurtbox/hurtbox.gd
+++ b/components/hurtbox/hurtbox.gd
@@ -3,13 +3,15 @@ class_name Hurtbox
 
 signal hit(hitbox)
 @onready var health: Health
+@onready var knockback: Knockback
 
 @export var invulnerability_time: float = 0.15
 var invulnerable: bool = false
 var invulnerability_timer: float = 0.0
 
 func _ready():
-	health = owner.get_node("Health") as Health
+	health = owner.get_node("Components/Health") as Health
+	knockback = owner.get_node("Components/Knockback") as Knockback
 
 func _process(delta: float) -> void:
 	if not invulnerable:
@@ -24,6 +26,14 @@ func _on_area_entered(area: Area2D) -> void:
 		return
 	if invulnerable:
 		return
+
 	health.take_damage(area.damage)
+
+	if knockback:
+		var direction := (owner.global_position - area.global_position)
+		if direction.is_zero_approx():
+			direction = Vector2.RIGHT
+		knockback.apply(direction, area.knockback_force)
+
 	invulnerable = true
 	invulnerability_timer = invulnerability_time
diff --git a/components/knockback/knockback.gd b/components/knockback/knockback.gd
index bb8e28d..2c4205d 100644
--- a/components/knockback/knockback.gd
+++ b/components/knockback/knockback.gd
@@ -19,4 +19,5 @@ func _physics_process(delta):
 	
 
 func apply(direction: Vector2, force: float):
-	pass
+	current_force = direction.normalized() * force
+	is_active = true
